fidesctl/tests/core/test_evaluate.py

```diff
@@ -407,11 +382,51 @@ def test_compare_rule_to_declaration_none_false_hierarchical():
     result = evaluate.compare_rule_to_declaration(
         rule_types=["key_1_parent"],
         declaration_type_hierarchies=[["key_2"], ["key_3"], ["key_1", "key_1_parent"]],
-        rule_inclusion="NONE",
+        rule_match="NONE",
     )
     assert not result
 
 
+@pytest.mark.unit
+def test_compare_rule_to_declaration_other_true():
+    result = evaluate.compare_rule_to_declaration(
+        rule_types=["key_1"],
+        declaration_type_hierarchies=[["key_2"], ["key_1"]],
+        rule_match="OTHER",
+    )
+    assert {"key_2"} == result
+
+
+@pytest.mark.unit
+def test_compare_rule_to_declaration_other_true_hierarchical():
+    result = evaluate.compare_rule_to_declaration(
+        rule_types=["key_1_parent"],
+        declaration_type_hierarchies=[["key_2"], ["key_1", "key_1_parent"]],
+        rule_match="OTHER",
+    )
+    assert {"key_2"} == result
+
+
+@pytest.mark.unit
+def test_compare_rule_to_declaration_other_false():
+    result = evaluate.compare_rule_to_declaration(
+        rule_types=["key_1"],
+        declaration_type_hierarchies=[["key_2"], ["key_3"]],
+        rule_match="OTHER",
+    )
+    assert result
+
+
+@pytest.mark.unit
+def test_compare_rule_to_declaration_other_false_hierarchical():
+    result = evaluate.compare_rule_to_declaration(
+        rule_types=["key_1"],
+        declaration_type_hierarchies=[["key_2", "key_2_parent"], ["key_3"]],
+        rule_match="OTHER",
+    )
+    assert result
```

Comment:
These two tests should be testing the case where `compare_rule_to_declaration` returns an empty set. Something like this:

```
    result = evaluate.compare_rule_to_declaration(
        rule_types=["key_1", "key_3"],
        declaration_type_hierarchies=["key_1"], ["key_3"]],
        rule_match="OTHER",
    )
    assert not result
```

This would test that there is no violations when keys other than key_1 and key_3 are not used. 

hierarchical:
```
    result = evaluate.compare_rule_to_declaration(
        rule_types=["key_1", "key_3_parent"],
        declaration_type_hierarchies=["key_1"], ["key_3", "key_3_parent"]],
        rule_match="OTHER",
    )
    assert not result
```